Mathlib/Data/Real/EReal.lean

```diff
@@ -1227,6 +1525,95 @@ lemma left_distrib_of_nonneg {a b c : EReal} (ha : 0 ≤ a) (hb : 0 ≤ b) :
   nth_rewrite 1 [EReal.mul_comm]; nth_rewrite 2 [EReal.mul_comm]; nth_rewrite 3 [EReal.mul_comm]
   exact right_distrib_of_nonneg ha hb
 
+lemma coe_mul_add_of_nonneg {x : ℝ} (hx_nonneg : 0 ≤ x) (y z : EReal) :
+    x * (y + z) = x * y + x * z := by
+  by_cases hx0 : x = 0
+  · simp [hx0]
+  have hx_pos : 0 < x := hx_nonneg.lt_of_ne' hx0
+  induction y
+  · simp [EReal.coe_mul_bot_of_pos hx_pos]
+  · induction z
+    · simp [EReal.coe_mul_bot_of_pos hx_pos]
+    · norm_cast
+      rw [mul_add]
+    · simp only [coe_add_top, EReal.coe_mul_top_of_pos hx_pos]
+      rw [← EReal.coe_mul, EReal.coe_add_top]
+  · simp only [EReal.coe_mul_top_of_pos hx_pos]
+    induction z
+    · simp [EReal.coe_mul_bot_of_pos hx_pos]
+    · simp only [top_add_coe, EReal.coe_mul_top_of_pos hx_pos]
+      rw [← EReal.coe_mul, EReal.top_add_coe]
+    · simp [EReal.coe_mul_top_of_pos hx_pos]
+
+lemma add_mul_coe_of_nonneg {x : ℝ} (hx_nonneg : 0 ≤ x) (y z : EReal) :
+    (y + z) * x = y * x + z * x := by
+  simp_rw [EReal.mul_comm _ (x : EReal)]
+  exact EReal.coe_mul_add_of_nonneg hx_nonneg y z
+
+lemma top_mul_add_of_nonneg {x y : EReal} (hx : 0 ≤ x) (hy : 0 ≤ y) :
```

Comment:
This is [EReal.left_distrib_of_nonneg](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Real/EReal.html#EReal.left_distrib_of_nonneg)

Reply:
Thank you for your feedback! 
Since these changes come from #17087, upon which this PR depends, I am going to fix this in #17087 and then merge it here.